crates/interpreter/src/valid.rs

```diff
@@ -764,17 +792,48 @@ impl<'a, 'm> Expr<'a, 'm> {
         Ok(())
     }
 
-    fn br_label(&self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
+    fn stitch(&mut self, source: SideTableBranch, target: SideTableBranch) {
+        let SideTableBranch { parser, side_table } = source;
+        let entry = &mut self.side_table[side_table];
+        assert!(entry.is_none());
+        *entry = Some(SideTableEntryView {
+            delta_ip: target.delta_ip(parser),
+            delta_stp: target.delta_stp(side_table),
+            // TODO(dev/fast-interp): Compute the fields below.
+            val_cnt: 0,
+            pop_cnt: 0,
+        });
+    }
+
+    fn br_label(&mut self, l: LabelIdx) -> Result<ResultType<'m>, Error> {
         let l = l as usize;
         let n = self.labels.len();
         check(l < n)?;
-        let label = &self.labels[n - l - 1];
+        let target = self.branch();
+        let label = &mut self.labels[n - l - 1];
         Ok(match label.kind {
-            LabelKind::Block | LabelKind::If => label.type_.results,
-            LabelKind::Loop => label.type_.params,
+            LabelKind::Block | LabelKind::If(_) => {
+                label.branches.push(target);
+                self.side_table.push(None);
+                label.type_.results
+            }
+            LabelKind::Loop(SideTableBranch { parser, side_table }) => {
+                self.side_table.push(Some(SideTableEntryView {
+                    delta_ip: target.delta_ip(parser),
+                    delta_stp: target.delta_stp(side_table),
+                    // TODO(dev/fast-interp): Compute the fields below.
+                    val_cnt: 0,
+                    pop_cnt: 0,
+                }));
```

Comment:
I think it's better to reuse code (even if the compiler might not optimize to the same code).
```suggestion
            LabelKind::Loop(target) => {
                self.side_table.push(None);
                self.stitch(source, target);
```

Reply:
Borrow checker error: the `label` above results from a mutable borrow, and `self.stitch()` is a second one.